src/index.js

```diff
@@ -0,0 +1,73 @@
+// copied from https://github.com/howdyai/botkit version 0.0.5, original authors: @RafaelCosman and @guillaumepotier
+
+var redis = require('redis');
+
+/**
+ * botkit-storage-redis - Redis driver for Botkit
+ *
+ * @param  {Object} config (optional) For full list of valid redis options, see
+ *  https://github.com/NodeRedis/node_redis#options-is-an-object-with-the-following-possible-properties
+ * @property config.namespace {String} The namespace to use when storing entities. Defaults to 'botkit:store'
+ * @return {Object} Storage interface for Botkit
+ */
+module.exports = function(config) {
+    config = config || {};
+    config.namespace = config.namespace || 'botkit:store';
+
+    var storage = {},
+        client = redis.createClient(config), // could pass specific redis config here
+        methods = ['teams', 'users', 'channels'];
+
+    // Implements required API methods
+    for (var i = 0; i < methods.length; i++) {
+        storage[methods[i]] = getStorageObj(client, config.namespace + ':' + methods[i]);
+    }
+
+    return storage;
+};
+
+/**
+ * Function to generate a storage object for a given namespace
+ *
+ * @param {Object} client The redis client
+ * @param {String} namespace The namespace to use for storing in Redis
+ * @returns {{get: get, save: save, all: all, allById: allById}}
+ */
+function getStorageObj(client, namespace) {
+    return {
+        get: function(id, cb) {
+            client.hget(namespace, id, function(err, res) {
+                cb(err, res ? JSON.parse(res) : {});
+            });
+        },
+        save: function(object, cb) {
+            if (!object.id) {
+                return cb(new Error('The given object must have an id property'), {});
+            }
+
+            client.hset(namespace, object.id, JSON.stringify(object), cb);
+        },
+        all: function(cb, options) {
+            client.hgetall(namespace, function(err, res) {
+                if (err) {
+                    return cb(err, {});
+                }
+
+                if (res === null) {
+                    return cb(err, res);
```

Comment:
According to line 52, if `err` exists this line will not execute. Therefore err is equal to `null`. I suggest removing `err` from this line. The line `res===null` sends `cb(null, null)` to the callback. This line doesn't make any sense.